great_expectations/experimental/datasources/spark_datasource.py

```diff
@@ -0,0 +1,229 @@
+            # imports are done inline to prevent a circular dependency with core/batch.py
+            from great_expectations.core import IDDict
+            from great_expectations.core.batch import BatchDefinition
+
+            batch_definition = BatchDefinition(
+                datasource_name=self.datasource.name,
+                data_connector_name="experimental",
+                data_asset_name=self.name,
+                batch_identifiers=IDDict(request.options),
+                batch_spec_passthrough=None,
+            )
+
+            batch_metadata = copy.deepcopy(request.options)
+            batch_metadata["path"] = path
+
+            # Some pydantic annotations are postponed due to circular imports. This will set the annotations before we
+            # instantiate the Batch class since we can import them above.
+            Batch.update_forward_refs()
+            batch_list.append(
+                Batch(
+                    datasource=self.datasource,
+                    data_asset=self,
+                    batch_request=request,
+                    data=data,
+                    metadata=batch_metadata,
+                    legacy_batch_markers=markers,
+                    legacy_batch_spec=batch_spec,
+                    legacy_batch_definition=batch_definition,
+                )
+            )
+        self.sort_batches(batch_list)
+        return batch_list
+
+
+class SparkDatasource(Datasource):
+    # class attrs
+    asset_types: ClassVar[List[Type[DataAsset]]] = [CSVSparkAsset]
+
+    # instance attributes
+    type: Literal["spark"] = "spark"
+    name: str
+    assets: Dict[str, CSVSparkAsset] = {}
+
+    @property
+    def execution_engine_type(self) -> Type[ExecutionEngine]:
+        """Return the SparkDFExecutionEngine unless the override is set"""
+        from great_expectations.execution_engine.sparkdf_execution_engine import (
+            SparkDFExecutionEngine,
+        )
+
+        return SparkDFExecutionEngine
+
+    def add_csv_asset(
+        self,
+        name: str,
+        base_directory: PathStr,
+        regex: Union[str, re.Pattern],
+        order_by: Optional[BatchSortersDefinition] = None,
+    ) -> CSVSparkAsset:
+        """Adds a csv asset to this Spark datasource
+
+        Args:
+            name: The name of the csv asset
+            base_directory: base directory path, relative to which CSV file paths will be collected
+            regex: regex pattern that matches csv filenames that is used to label the batches
+            order_by: one of "asc" (ascending) or "desc" (descending) -- the method by which to sort "Asset" parts.
+        """
+        asset = CSVSparkAsset(
+            name=name,
+            base_directory=base_directory,  # type: ignore[arg-type]  # str will be coerced to Path
+            regex=regex,  # type: ignore[arg-type]  # str with will coerced to Pattern
+            order_by=order_by or [],  # type: ignore[arg-type]  # coerce list[str]
```

Comment:
Are all these `type: ignore` comments necessary? Is there a way to get around it?

Reply:
@cdkini I will check (they were in the `Pandas` version).  Thanks.

Reply:
@cdkini I checked.  Without these "ignore" comments, the following happened:

```
mypy great_expectations/experimental/datasources/spark_datasource.py
great_expectations/experimental/datasources/spark_datasource.py:225: error: Argument "base_directory" to "CSVSparkAsset" has incompatible type "Union[str, Path]"; expected "Path"  [arg-type]
great_expectations/experimental/datasources/spark_datasource.py:226: error: Argument "regex" to "CSVSparkAsset" has incompatible type "Union[str, Pattern[Any]]"; expected "Pattern[Any]"  [arg-type]
great_expectations/experimental/datasources/spark_datasource.py:227: error: Argument "order_by" to "CSVSparkAsset" has incompatible type "Union[List[BatchSorter], List[str], List[<nothing>]]"; expected "List[BatchSorter]"  [arg-type]
Found 3 errors in 1 file (checked 1 source file)
```

With these "ignores" in place, `mypy` ran cleanly.

Thanks.